src/common/pf_alarm.c

```diff
@@ -478,32 +478,29 @@ static int pf_alarm_apmr_high_handler(
    uint16_t                frame_id_pos,
    void                    *p_arg)
 {
-   int                     ret = 1;       /* Means "handled" */
    pf_apmx_t               *p_apmx = (pf_apmx_t *)p_arg;
    pf_apmr_msg_t           *p_apmr_msg;
    uint16_t                nbr;
 
-   if (ret != 0)
+   LOG_INFO(PF_ALARM_LOG, "Alarm(%d): Received high prio alarm frame.\n", __LINE__);
+
+   if (p_buf != NULL)
    {
-      LOG_INFO(PF_ALARM_LOG, "Alarm(%d): Received high prio alarm frame.\n", __LINE__);
-      if (p_buf != NULL)
+      nbr = p_apmx->apmr_msg_nbr++;    /* ToDo: Make atomic */
+      if (p_apmx->apmr_msg_nbr >= NELEMENTS(p_apmx->apmr_msg))
       {
-         nbr = p_apmx->apmr_msg_nbr++;    /* ToDo: Make atomic */
-         if (p_apmx->apmr_msg_nbr >= NELEMENTS(p_apmx->apmr_msg))
-         {
-            p_apmx->apmr_msg_nbr = 0;
-         }
-         p_apmr_msg = &p_apmx->apmr_msg[nbr];
-         p_apmr_msg->p_buf = p_buf;
-         p_apmr_msg->frame_id_pos = frame_id_pos;
-         if (os_mbox_post(p_apmx->p_alarm_q, (void *)p_apmr_msg, 0) != 0)
-         {
-            LOG_ERROR(PF_ALARM_LOG, "Alarm(%d): Lost one alarm\n", __LINE__);
-         }
+         p_apmx->apmr_msg_nbr = 0;
+      }
+      p_apmr_msg = &p_apmx->apmr_msg[nbr];
+      p_apmr_msg->p_buf = p_buf;
+      p_apmr_msg->frame_id_pos = frame_id_pos;
+      if (os_mbox_post(p_apmx->p_alarm_q, (void *)p_apmr_msg, 0) != 0)
+      {
+         LOG_ERROR(PF_ALARM_LOG, "Alarm(%d): Lost one alarm\n", __LINE__);
       }
    }
 
-   return ret;
+   return 1;  /* Means that frame handler should not free buffer, as that will be done when reading the mbox */
```

Comment:
If alarm is not added to mbox (the error  case), is that a memory leak?  Maybe p_buf shall be freed on mbox_post error?

Reply:
Good point. I have not really modified the code. We need to look into how the mbox handles error cases.

Reply:
Done.